transport/src/test/java/io/camunda/zeebe/transport/stream/impl/ClientStreamManagerTest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.transport.stream.impl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.atomix.cluster.MemberId;
+import io.atomix.cluster.messaging.ClusterCommunicationService;
+import io.camunda.zeebe.scheduler.testing.TestConcurrencyControl;
+import io.camunda.zeebe.util.buffer.BufferReader;
+import io.camunda.zeebe.util.buffer.BufferUtil;
+import io.camunda.zeebe.util.buffer.BufferWriter;
+import java.util.concurrent.CompletableFuture;
+import org.agrona.DirectBuffer;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class ClientStreamManagerTest {
+
+  private final DirectBuffer streamType = BufferUtil.wrapString("foo");
+  private final BufferWriter metadata = mock(BufferWriter.class);
+  private final ClientStreamRegistry<BufferWriter, BufferReader> registry =
+      new ClientStreamRegistry<>();
+  private final ClusterCommunicationService mockTransport = mock(ClusterCommunicationService.class);
+  private final ClientStreamManager<BufferWriter, BufferReader> clientStreamManager =
+      new ClientStreamManager<>(
+          registry, new ClientStreamRequestManager<>(mockTransport, new TestConcurrencyControl()));
```

Comment:
:thought_balloon: I think this highlights how useful splitting the `ClusterCommunicationService` up would help :)